### ISSUE TITLE
haxe.macro.Field.doc only recognizes /** doc */, no /// doc

### ISSUE THREAD

**Issue by YellowAfterlife** [HaxeFoundation/haxe#8164]
As of Apr 12 build,
```haxe
@:keep @:build(Macro.build())
class Test {
  /// one!
  static function one() {}
  /** two */
  static function two() {}
  
  static function main() {
    trace("Haxe is great!");
  }
}
```
```haxe
//use this for macros or other classes
import haxe.macro.Expr;
import haxe.macro.Context;

class Macro {
  public static function build() {
    var fields = Context.getBuildFields();
    for (field in fields) Context.warning("" + field.doc, field.pos);
    return null;
	}
}
```
gets you
```
Test.hx:4: characters 3-27 : Warning : null
Test.hx:6: characters 3-27 : Warning :  two 
```
It'd be lovely if `///` was also allowed for docs, as it is convenient to pull out docs into a quick reference page through a macro, but not so much when you get to choose between odd-looking
```haxe
/** Queued up input states (for local slots, size matches input delay) */
public static var states:Vector<ArrayList<InputState>>;
```
and bulkier (3 lines of code for 1 line of text)
```haxe
/**
    Queued up input states (for local slots, size matches input delay)
**/
public static var states:Vector<ArrayList<InputState>>;
```

**Comment by Gama11** [comment]
I would disagree that `/** */` in a single line is odd-looking, though the option I see more frequently is `/** **/`. Either way, I don't think Haxe needs a new "comment type".

**Comment by YellowAfterlife** [comment]
FlashDevelop/HaxeDevelop supported `///` comments as a distinct token type (`commentlinedoc`) since introduction of support for Haxe, and triple-slash comments are used either for documentation or metadata in variety of other languages/editors/tools ([TS](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html), [GML](http://docs2.yoyogames.com/source/_build/1_overview/3_additional_information/jsdoc.html), [C#](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/recommended-tags-for-documentation-comments), [doxygen](http://www.doxygen.nl/manual/docblocks.html), etc.) so I wouldn't consider it particularly wild to support.

**Comment by Gama11** [comment]
IMO it's not correct for FD to support that right now, since it's not really part of the language. But that's a different discussion I guess.

**Comment by back2dos** [comment]
It's not like FD invented this comment style and it is a bit thin to argue about comments not really being part of the language, because that's the nature of comments. What's more, [none](https://github.com/HaxeFoundation/dox/wiki/Generating-documentation-XML-files-using-Haxe) [of](https://api.haxe.org/haxe/macro/Field.html#doc) [the](https://api.haxe.org/haxe/macro/ClassField.html#doc) [docs](https://haxe.org/manual/class-field.html) actually specify the syntax for inline documentation. Now, I'm sure there are countless creative reasons why this shouldn't be supported, like calling an inline documentation style "new" that has been in use since before "the language" was a thing. Or: we could simply say that it was an oversight to not support this widely used type of inline documentation, fix it and move on ;)

**Comment by nadako** [comment]
My problem with this is that people for sure will use this as a multi-line documentation comment, which is stupid, but hey, C# supports that. So we'll have to collect multiple tokens into one etc.

**Comment by Simn** [comment]
We should document that `/** documentation */` is the supported documentation style, but I see no reason to add another syntax for the same thing.
